drivers/dma/dma_silabs_ldma.c

```diff
@@ -0,0 +1,564 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdbool.h>
+#include <stddef.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/mem_blocks.h>
+
+#include "em_ldma.h"
+
+#define DT_DRV_COMPAT silabs_ldma
+
+#define DMA_IRQ_PRIORITY 3
+
+LOG_MODULE_REGISTER(silabs_dma, CONFIG_DMA_LOG_LEVEL);
+
+struct dma_silabs_channel {
+	uint8_t id;
+	enum dma_channel_direction dir;
+	uint32_t complete_callback_en;
+	volatile bool busy;
+	void *user_data;
+	dma_callback_t cb;
+	LDMA_TransferCfg_t xfer_config;
+	LDMA_Descriptor_t *desc;
+};
+
+struct dma_silabs_config {
+	void (*config_irq)(const struct device *dev);
+	const struct device *clock_dev;
+};
+
+struct dma_silabs_data {
+	struct dma_context dma_ctx;
+	struct dma_silabs_channel *dma_chan_table;
+	struct sys_mem_blocks *dma_desc_pool;
+};
+
+struct table {
+	int native;
+	int efr;
+};
+
+const static struct table ldma_blocksize_map[] = {
+	{0x1, ldmaCtrlBlockSizeUnit1},     {0x2, ldmaCtrlBlockSizeUnit2},
+	{0x3, ldmaCtrlBlockSizeUnit3},     {0x4, ldmaCtrlBlockSizeUnit4},
+	{0x6, ldmaCtrlBlockSizeUnit6},     {0x8, ldmaCtrlBlockSizeUnit8},
+	{0x10, ldmaCtrlBlockSizeUnit16},   {0x20, ldmaCtrlBlockSizeUnit32},
+	{0x40, ldmaCtrlBlockSizeUnit64},   {0x80, ldmaCtrlBlockSizeUnit128},
+	{0x100, ldmaCtrlBlockSizeUnit256}, {0x200, ldmaCtrlBlockSizeUnit512},
+	{0x400, ldmaCtrlBlockSizeUnit1024}};
+
+const static struct table ldma_prio_map[] = {{0x0, ldmaCfgArbSlotsAs1},
```

Comment:
These values are related by `efr =  BIT(native)`, meaning that table lookup isn't strictly necessary.

Reply:
My comment was for `ldma_prio_map`, which you already simplified. Agree that block size requires a lookup table.